### ISSUE TITLE
Relax reference name validation with ValidationStringency

### ISSUE THREAD

**Issue by maxibor** [Integrative-Transcriptomics/DamageProfiler#39]
Right now, the (default ?) reference name validation stringency of htsjdk is pretty strict, leading to errors when reference names in alignment files are ill-formated (for example, the refererence names in the [metaphlan](https://github.com/biobakery/metaphlan) database).
This should be relaxed with [ValidationStringency](https://www.javadoc.io/doc/com.github.samtools/htsjdk/latest/htsjdk/samtools/ValidationStringency.html) to allow for non-perfectly formatted reference names.
CC @apeltzer @JudithNeukamm

**Comment by JudithNeukamm** [comment]
Thanks for your comment. 
The ValidationStringency is set to LENIENT by default which emits warnings but keeps the run going if possible. Did you get a wrong output or did the tool just throw a warning?

**Comment by maxibor** [comment]
```bash
$ damageprofiler -i metagenomebis.all_mapped.bam -r mpa_db_latest.fa -o damageprofiler
DamageProfiler v0.4.6
Invalid SAM/BAM file. Please check your file.
htsjdk.samtools.SAMException: Sequence name '157592__A0A150IGK6__fliD,flbC,flaV' doesn't match regex: '[0-9A-Za-z!#$%&+./:;?@^_|~-][0-9A-Za-z!#$%&*+./:;=?@^_|~-]*'
```

Version: 0.4.6, installed via Conda
Test files are unfortunately too big to attach. The error is with the characters in the sequence name, not allowed by the regex (in the example above, a `,`)

**Comment by JudithNeukamm** [comment]
I did a small test file, and changing the ValidationStringency to 'SILENT' does not solve the problem, unfortunately. I will try to solve this until the next release.

**Comment by apeltzer** [comment]
To be honest, thats also a quite invalid FastA header 🙄 `157592__A0A150IGK6__fliD,flbC,flaV` 🤦

**Comment by maxibor** [comment]
Agree, Metaphlan uses funny reference names. Though, for example, this is valid: `157592__A0A150IGK6__fliD;flbC;flaV`

**Comment by JudithNeukamm** [comment]
Unfortunately, I couldn't solve this problem. It's not influenced by the ValidationStringency parameter, and there doesn't seem to be an option to set a user-defined regex pattern. 
It might be an option to contact the developer of metaphlan to make them aware of this issue? Or to fix the file before running DaamageProfiler. 
If you find any solution to solve this within the code, I'll happy to include it.

**Comment by terrycojones** [comment]
Hi all. I ran into this today, too. An additional comment/request is to make the error output more informative. On lines 90-94 of you have this:

```java
             } catch (Exception e){
                  System.err.println("Invalid SAM/BAM file. Please check your file.");
                   LOG.error("Invalid SAM/BAM file. Please check your file.");
                   System.exit(-1);
               }
  ```

I know almost zero Java, but surely you could print the error information in the exception? To figure out what the problem was, I had to do many things:

- find my way to this repo
- clone it
- search in the source code (to find the above)
- then I went looking online and saw someone recommending checking the BAM file with picard
- I checked the BAM file with samtools quickcheck (no problem)
- Searched for and downloaded the gatk jar file
- Ran the command to check the file and it told me the reference names were syntactically invalid

I think that if you'd just printed the exception I would have known straight away.

Anyway, thanks for DamageProfiler, it looks very useful. Now I'll go rewrite my sequence ids..... :-)
